src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -15,6 +17,7 @@ public class ParamsSemanticModel
         public Compilation? BicepCompilation { get; }
         public ParamsTypeManager ParamsTypeManager { get; }
         public ParamsSymbolContext ParamsSymbolContext { get; }
+        private List<IDiagnostic> semanticDiagnostics = new List<IDiagnostic>();
```

Comment:
We're keen to avoid any mutable state on our classes. I think you should be able to avoid this by just using a local variable in `GetAdditionalSemanticDiagnostics`.